src/Conversion/ONNXToTOSA/Tensor/Resize.cpp

```diff
@@ -0,0 +1,294 @@
+
+    if (inputShape[0] != outputShape[0] || inputShape[1] != outputShape[1])
+      return rewriter.notifyMatchFailure(
+          op, "Cannot resize non spatial dimensions.");
+
+    // Get axis values if set. Default is all axis in normal order.
+    llvm::SmallVector<int64_t> axisVec;
+    if (axis.has_value()) {
+      valuesFromAxis(&axis.value(), axisVec);
+    } else {
+      axisVec.append({0, 1, 2, 3});
+    }
+
+    // Set these explicitly just out of convinience.
+    int64_t inputHeight = inputShape[2];
+    int64_t inputWidth = inputShape[3];
+    int64_t outputHeight = outputShape[2];
+    int64_t outputWidth = outputShape[3];
+
+    // Check if scales are set. We need to get those float values, because they
+    // make a difference in linear interpolation.
+    Value scaleValue = resizeOp.getScales();
+    llvm::SmallVector<float, 4> scales{1, 1, 1, 1};
+    if (!isNoneValue(scaleValue)) {
+      auto elementsAttr = getElementAttributeFromONNXValue(scaleValue);
+      if (!elementsAttr)
+        return rewriter.notifyMatchFailure(
+            op, "Scale cannot come from a block argument.");
+
+      // The axis attribute might permute and/or reduce the number of elements
+      // in scaleValue. This reorders the scales to account for that.
+      for (auto [index, value] : llvm::enumerate(axisVec))
+        scales[value] =
+            elementsAttr.getValues<FloatAttr>()[index].getValueAsDouble();
+
+      // Even if the shapes are identical, a scale value other than 1 is
+      // possible. TOSA does not allow that for non-spatial dimensions.
+      for (int64_t nonSpatialDimensions : {0, 1}) {
+        if (scales[nonSpatialDimensions] != 1)
+          return rewriter.notifyMatchFailure(
+              op, "Axis Attr if present must contain both output dimesions.");
+      }
+
+      // In TOSA the scale is a fraction of two integer numbers.
+      std::tie(outputHeight, inputHeight) = floatToFraction(scales[2]);
+      std::tie(outputWidth, inputWidth) = floatToFraction(scales[3]);
+    }
+
+    bool alignCorners = coordinateTransformationMode == "align_corners";
+    bool halfPixel = coordinateTransformationMode == "half_pixel";
+    bool pytorchHalfPixel =
+        coordinateTransformationMode == "pytorch_half_pixel";
+    bool halfPixelSymmetric =
+        coordinateTransformationMode == "half_pixel_symmetric";
+    bool isBilinear = mode == "linear";
+    bool isNearest = mode == "nearest";
+    bool isNearestModeFloor = nearestMode == "floor";
+    StringRef resizeMode = isBilinear ? "BILINEAR" : "NEAREST_NEIGHBOR";
+
+    if (halfPixelSymmetric)
+      return rewriter.notifyMatchFailure(op,
+          "TOSA does not support float offsets which are required "
+          "for symmetric mode.");
+
+    // Adapted from TFL to TOSA.
+    auto normalize = [&](int64_t output, int64_t input) {
+      // Dimension is length 1, we are just sampling from one value.
+      int64_t n, d, offset, border;
```

Comment:
Could we have better naming than n and d there, makes the code quite harder to read when you don't know what they are about